youtube_dl/extractor/funimation.py

```diff
@@ -0,0 +1,97 @@
+# coding: utf-8
+from __future__ import unicode_literals
+from .common import InfoExtractor
+from ..compat import compat_HTTPError
+from ..utils import (
+    sanitized_Request,
+    ExtractorError,
+    urlencode_postdata
+)
+
+import re
+
+class FunimationIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?funimation\.com/shows/.*/videos/official/(?P<id>[^?]+)'
+    _USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/601.2.7 (KHTML, like Gecko) Version/9.0.1 Safari/601.2.7'
+
+    _TEST = {
+        'url': 'http://www.funimation.com/shows/air/videos/official/breeze',
+        'info_dict': {
+            'id': 'AIRENG0001',
+            'title': 'Air - Breeze',
+            'ext': 'mp4',
+        },
+        'params': {
+            'format': 'lower',
+        }
+    }
+
+    def _real_initialize(self):
+        self._login()
+
+    def _login(self):
+        (username, password) = self._get_login_info()
+        if username is None:
+            return
+        self.report_login()
+        login_url = 'http://www.funimation.com/login'
+        data = urlencode_postdata({
+            'formname': 'loginForm2',
+            'email_field': username,
+            'password_field': password,
```

Comment:
All fields that may potentially contain non-ASCII must be [utf-8 encoded](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/vimeo.py#L43).